make/clang-tidy

```diff
@@ -21,3 +21,7 @@ clang-tidy-fix:
 		 -checks='$(tidy_checks)' -format-style='file' -header-filter='./stan/*' \
 		 -fix -export-fixes="./.clang-fixes.log" \
 		 -- $(CXXFLAGS) $(LDFLAGS) $(LDLIBS) $(INC_GTEST) $(CXXFLAGS_GTEST) $(CPPFLAGS)
+
+.PHONY: clang-format
+clang-format:
+		find stan test -name '*.hpp' -o -name '*.cpp' | xargs -n20 -P1 clang-format-5.0 -i
```

Comment:
Was this intended to be included in this PR?

Reply:
So yes, mostly because I wanted to include it for people in the contributors guide so they could format their code using `make clang-format`. I was going to make another PR but it's only 4 lines so it felt 'fine-ish'

Reply:
Are there links for installing our makefile stuff somewhere? I can link to that in the contributing.md

Reply:
https://github.com/stan-dev/stan/wiki/Coding-Style-and-Idioms#automated-style-checking-tools

Also we suggest they install the pre-commit hook, which would mean they don't need a makefile target. That target reformats the world instead of just the files that someone changed.